salt/grains/core.py

```diff
@@ -983,6 +984,59 @@ def _virtual(osdata):
         )
     return grains
 
+def _virtual_hv(osdata):
+    '''
+    Returns detailed hypervisor information from sysfs
+    Currently this seems to be used only by Xen
+    '''
+    grains = {}
+
+    # Bail early if we're not running on Xen
+    if 'xen' not in osdata['virtual']:
+        return grains
+
+    # Try to get the exact hypervisor version from sysfs
+    try:
+        version = {}
+        for file in ('major', 'minor', 'extra'):
```

Comment:
`file` is a reserved keyword in Python. Can you change this to another variable name?

Reply:
Done.